### ISSUE TITLE
"The given key was not present in the dictionary." BGI error

### ISSUE THREAD

**Issue by Linley0** [arcusmaximus/VNTranslationTools#17]
I got an error when extracting script from .BGI
"The given key was not present in the dictionary."
I wonder why this happen?

**Comment by arcusmaximus** [comment]
Please attach the script file and let me know what game this is for so I can investigate.

**Comment by Linley0** [comment]
[bgi script.zip](https://github.com/arcusmaximus/VNTranslationTools/files/7801934/bgi.script.zip)

the game : https://vndb.org/v7766

**Comment by Linley0** [comment]
Also get the same error with other game 
[bgi script.zip](https://github.com/arcusmaximus/VNTranslationTools/files/7805530/bgi.script.zip)
https://vndb.org/v16342

**Comment by arcusmaximus** [comment]
Looks like 11月のアルカディア uses quite a few scenario opcodes that VNTextPatch doesn't know about yet. I'd need to disassemble the scenario VM to see what kind of parameters these opcodes take, but it's proving difficult to acquire the game. Could you also attach the ._bp scripts?

**Comment by xenei** [comment]
Was about to make a issue about this but then I saw someone already made one, but they got that error with some BGI vns. Me, however, got the same error with one AdvHD vn, here's what it shows - (same issue even if I try extract to script.xlsx)
![sda](https://user-images.githubusercontent.com/91423244/149349043-3c0ddf5b-2369-46a0-b74e-81593e3fb9ba.PNG)
Initially I thought I needed to give some key but then I remembered there was no note to give any key. I just wanted to give you a notice about this issue after the recent implement of AdvHD engine support.
Here's the extracted .ws2 files extracted with GarBro along the with RIO1.arc and RIO.arc file, And the VN that I got the error with is - [https://vndb.org/v28511](url)
[AdvHD.zip](https://github.com/arcusmaximus/VNTranslationTools/files/7863545/AdvHD.zip)

**Comment by Linley0** [comment]
> Looks like 11月のアルカディア uses quite a few scenario opcodes that VNTextPatch doesn't know about yet. I'd need to disassemble the scenario VM to see what kind of parameters these opcodes take, but it's proving difficult to acquire the game. Could you also attach the ._bp scripts?

Sorry for late reply.. here's the .arc file with ._bp script inside
https://drive.google.com/file/d/1hPXK2utziwXwyrBMD0v2arfFX0tQuGbD/view?usp=sharing

**Comment by KuronekoRM** [comment]
I've got the same issue with an AdvHD engine game, Dynamic Chord apple-polisher [https://vndb.org/v20072](https://vndb.org/v20072).
It seems there are only 2 .ws2 files that can't be extracted: aptoi_38.ws2 and sp05_boys.ws2. I took those two out of the folder and the rest were extracted without a problem, I could even edit them and reinsert them. Just not these 2.
Here's the Rio.arc and the extracted ws2 files (which I extracted using Garbro)
[Script.zip](https://github.com/arcusmaximus/VNTranslationTools/files/7873492/Script.zip)
